weaver/wps_restapi/processes/processes.py

```diff
@@ -136,27 +177,33 @@ def execute_process(self, job_id, url, headers=None, notification_email=None):
             if num_retries >= max_retries:
                 raise Exception("Could not read status document after {} retries. Giving up.".format(max_retries))
             try:
-                execution = check_status(url=execution.statusLocation, verify=ssl_verify,
-                                         sleep_secs=wait_secs(run_step))
-
+                # NOTE:
+                #   Don't actually log anything here until process is completed (success or fail) so that underlying
+                #   WPS execution logs can be inserted within the current job log and appear continuously.
+                #   Only update internal job fields in case they get referenced elsewhere.
```

Comment:
And how the underlying wps execution can do that?

Reply:
Everything that gets logged to file is retrieved form the execution of 
 https://github.com/crim-ca/weaver/blob/7a6bed9ad0c847fa7878f75737f2629e3a2f58ae/weaver/processes/wps_package.py#L1612

So for the moment, we still don't get much more async update than before *during* the internal execution of the process because we retrieve it *after* its done. But, we get more feedback for other steps. 

For example, we had something like :
```
[0] Job started.
<nothing for a while>
[1..N-1] <potentially some error message (depends of what WPS reports back)>
[N] Job failed.
```
Now something more detailed : 
```
[0] Job started.
[1] Setup this.
[2] Setup that.
[3] Process monitor started 
(we know at least the error is WPS side now - not db/worker issue).
<nothing for a while>
[4..N-2] <potentially some error message (depends of what WPS reports back)>
[N-1] Oh oh, something bad happened during X.
[N] Job failed.
```

The note is mostly to remember that, if we add monitoring logs during `<nothing for a while>`, the order of `[#]` log entries will make the traceback unreadable.